eventuate-common-quarkus-in-memory-database/src/main/java/io/eventuate/common/quarkus/inmemorydatabase/EventuateCommonInMemoryDatabaseConfiguration.java

```diff
@@ -0,0 +1,22 @@
+package io.eventuate.common.quarkus.inmemorydatabase;
+
+import io.eventuate.common.inmemorydatabase.EventuateDatabaseScriptSupplier;
+import io.eventuate.common.inmemorydatabase.EventuateInMemoryDataSourceBuilder;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Alternative;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.Produces;
+import javax.sql.DataSource;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+@Alternative
+@Priority(0)
+public class EventuateCommonInMemoryDatabaseConfiguration {
+  @Produces
+  public DataSource dataSource(Instance<EventuateDatabaseScriptSupplier> scripts) {
+    return new EventuateInMemoryDataSourceBuilder(scripts.stream().collect(Collectors.toList())).build();
```

Comment:
Is this code that's duplicated elsewhere?

Reply:
For micronaut/spring in is different.
micronaut: https://github.com/eventuate-foundation/eventuate-common/blob/wip-db-id-gen/eventuate-common-micronaut-in-memory-database/src/main/java/io/eventuate/common/micronaut/inmemorydatabase/EventuateCommonInMemoryDatabaseFactory.java
spring: https://github.com/eventuate-foundation/eventuate-common/blob/wip-db-id-gen/eventuate-common-spring-in-memory-database/src/main/java/io/eventuate/common/spring/inmemorydatabase/EventuateCommonInMemoryDatabaseConfiguration.java